### PR TITLE
Bring upstream changes

### PR DESCRIPTION
**Pull request by ivangonzalezacuna**
I've finally been able to rebase the changes from upstream.

I'm testing it in the branch with the upgraded backstage 1.12.1 and it works.

Unfortunately, we cannot fully use the upstream plugin. The reason is the custom dashboards we are adding. With the new tag-selector formatting we could select the dashboards by name (by overwriting the annotation in our processors), which is working fine. However, we are losing the option of adding several parameters to that URL, so this is not fully supported yet.

To note as well. The state of the new alerts are still not shown, however there a PR already in place, so I guess it could be ready at some point soon

### CODE REVIEW

**Review by heyLu — CHANGES_REQUESTED** [review]
Regarding this message in 21a1809f1caba190a6983b7298864ee59f0699bd:

>  Add again hook to inject other dashboards
>
> This is not fully supported. I could overwrite the annotation to always show the 2/3 special boards,
> but I cannot customize the parameters, therefore the user would need to search for the correct component
> to display the same data as it's shown now

I am not sure what this means: Does this mean that _upstream_ does not support adding custom parameters, like we do internally using the feature from this commit?  As far as I can tell this adds back the feature that we need internally and we can add the parameters with this feature again now, even with the rest of the code from upstream?

**Comment by heyLu** [comment]
Did this need any changes to the code from upstream or are the only "new" things added the last two commits?  (Apart from the upstream changes.)

**Comment by ivangonzalezacuna** [comment]
> Regarding this message in [21a1809](https://github.com/spreadshirt/backstage-plugin-grafana/commit/21a1809f1caba190a6983b7298864ee59f0699bd):
> 
> > Add again hook to inject other dashboards
> > This is not fully supported. I could overwrite the annotation to always show the 2/3 special boards,
> > but I cannot customize the parameters, therefore the user would need to search for the correct component
> > to display the same data as it's shown now
> 
> I am not sure what this means: Does this mean that _upstream_ does not support adding custom parameters, like we do internally using the feature from this commit? As far as I can tell this adds back the feature that we need internally and we can add the parameters with this feature again now, even with the rest of the code from upstream?

What I mean is that, now, we can add/customize the annotation and make the plugin fetch more dashboards. For example, setting the annotation to `grafana/dashboard-selector: "tags @> 'backstage' || title == 'k8s-Pods (new clusters)'"`, the plugin will also fetch that board (and this can be easily customizable in our enrich processor by adding some expressions. We could even use the url to be sure that we're fetching the correct board, define in which board they are and so on). 

However, we are currently customizing a bit more that link. In the example above, we are setting a query param with the service name, so if you open it, it will redirect you to the data for only that service. For example, for backstage: https://grafana.sprd.net/d/N9PNo1QZz/k8s-pods-new-clusters?orgId=1&var-deployment=backstage&var-pod=All (**NOTE the var-deployment=backstage field**).

Since this is not fully supported yet, we have to remain using our implementation.

So basically, the 2 commits I added are:
* Upgrade dependencies to 1.12.1 again (some of them are not updated in the upstream plugin)
* Add the integration for additionalDashboards again

**Review by heyLu — APPROVED** [review]
